### DISCUSSION TITLE
nvim-treesitter not recognizes filetype `sh`.

### DISCUSSION THREAD

**Discussion by charliie-dev** [ayamir/nvimdots#272] [Q&A] [answered]
For some reason, nvim-treesitter doesn't pick up filetype `sh` as `bash`. It cause some issue for some plugins related to treesitter.
I have tried few attempts to change filetype via **ftdetect** or **autocommand**, but there're some issues stated in #270 .

I also tried adding `sh` to treesitter bash parser:

```lua
local parser_config = require("nvim-treesitter.parsers").get_parser_configs()
parser_config.bash.filetype_to_parsername = {
	"sh", --[["zsh"--]]
}
```
but it didn't work either.

Is there any other way to make filetype `sh` use treesitter `bash` parser?

**Comment by ayamir** [comment]
highlight is enabled for me by default.
Enabled:
![image](https://user-images.githubusercontent.com/61657399/194826005-cc0fa5eb-365c-4c8d-9ba9-5be3958eefdb.png)
Disabled:
![image](https://user-images.githubusercontent.com/61657399/194826114-e73322e5-69f8-4414-8d68-7c57ba184043.png)

**Comment by charliie-dev** [comment]
@ayamir ys, highlight's are available for me too.
![image](https://user-images.githubusercontent.com/32497323/194826675-ba2730a9-951b-4d01-937a-3d4d35c6fcfb.png)

But when I tried to use this treesitter releted plugin: [`s1n7ax/nvim-comment-frame`](https://github.com/s1n7ax/nvim-comment-frame)

It popout error:
![image](https://user-images.githubusercontent.com/32497323/194827201-a4802baa-8109-4f3a-9025-fee5f81541a6.png)

It could be fixed by manually execute `:set filetype=bash` when editting in a `sh` filetype buffer, and the plugin will work.

**Reply by charliie-dev** [reply]
Install nvim-comment-frame:
```lua
editor["s1n7ax/nvim-comment-frame"] = {
	opt = true,
	event = "BufReadPost",
	requires = {
		{ "nvim-treesitter/nvim-treesitter", opt = true },
	},
	config = function()
		require("modules.editor.comment-frame")
	end,
```

comment-frame.lua:
```lua
require("nvim-comment-frame").setup({
	-- width of the comment frame
	frame_width = 70,

	-- wrap the line after 'n' characters
	line_wrap_len = 50,

	-- automatically indent the comment frame based on the line
	auto_indent = true,

	-- add comment above the current line
	add_comment_above = true,

	languages = {
		-- configuration for Lua programming language
		-- @NOTE global configuration will be overridden by language level
		-- configuration if provided
		lua = {
			start_str = "--",
			end_str = "--",
			fill_char = "-",
		},
		bash = {
			start_str = "#",
			end_str = "#",
			fill_char = "#",
		},
		sh = {
			start_str = "#",
			end_str = "#",
			fill_char = "#",
		},
		python = {
			start_str = "#",
			end_str = "#",
			fill_char = "#",
		},
	},
})
```

**Reply by charliie-dev** [reply]
I also tried to add extra context_commentstring config, but it doesn't work either.

```lua
require("nvim-treesitter.configs").setup({
	context_commentstring = {
		enable = true,
		enable_autocmd = false,
		config = {
			sh = "# %s",
		},
	},
})
```

**Comment by charliie-dev** [comment]
Probably related to [this](https://github.com/s1n7ax/nvim-comment-frame/blob/05be8b4cf2138d341ed5135dc704d9fccf51a3cb/lua/nvim-comment-frame/init.lua#L48-L50) ?

**Reply by ayamir** [reply]
Maybe, you can wait this plugin's author fix it or set `sh` for `bashrc` etc explictly to solve this problem temporarily.

**Reply by charliie-dev** [reply]
Actucally, `sh` is the detected filetype for `bashrc`, and it's not working with "s1n7ax/nvim-comment-frame" unless manually setting the filetype to `bash` after entering the buffer.

I also asked the plugin author about this issue: https://github.com/s1n7ax/nvim-comment-frame/issues/10, currently waiting for response.

**Comment by charliie-dev** [comment, marked as answer]
The author of the plugin has fix this https://github.com/s1n7ax/nvim-comment-frame/issues/10.
